matlab/src/cpp/arrow/matlab/tabular/proxy/schema.cc

```diff
@@ -0,0 +1,196 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/matlab/error/error.h"
+#include "arrow/matlab/tabular/proxy/schema.h"
+#include "arrow/matlab/type/proxy/field.h"
+
+#include "libmexclass/proxy/ProxyManager.h"
+#include "libmexclass/error/Error.h"
+
+#include "arrow/util/utf8.h"
+
+#include <sstream>
+
+namespace arrow::matlab::tabular::proxy {
+
+    namespace {
+
+        libmexclass::error::Error makeUnknownFieldNameError(const std::string& name) {
+            using namespace libmexclass::error;
+            const std::string error_message_id = std::string{error::ARROW_TABULAR_SCHEMA_UNKNOWN_FIELD_NAME};
+            std::stringstream error_message_stream;
+            error_message_stream << "Unknown field name: '";
+            error_message_stream << name;
+            error_message_stream << "'.";
+            const std::string& error_message = error_message_stream.str();
+            return Error{error_message_id, error_message};
+        }
+
+        libmexclass::error::Error makeEmptySchemaError() {
+            using namespace libmexclass::error;
+            const std::string error_message_id = std::string{error::ARROW_TABULAR_SCHEMA_NUMERIC_FIELD_INDEX_WITH_EMPTY_SCHEMA};
+            std::stringstream error_message_stream;
+            error_message_stream << "Numeric indexing using the field method is not supported for schemas with no fields.";
+            const std::string& error_message = error_message_stream.str();
+            return Error{error_message_id, error_message};
```

Comment:
Can we simplify this (and reduce needless memory allocations)?

```suggestion
            return Error{error::ARROW_TABULAR_SCHEMA_NUMERIC_FIELD_INDEX_WITH_EMPTY_SCHEMA,
                         "Numeric indexing using the field method is not supported for schemas with no fields."};
```